services/blockchain-indexer/app.js

```diff
@@ -51,31 +51,63 @@ const app = Microservice({
 	],
 });
 
-setAppContext(app);
+const app = Microservice(defaultBrokerConfig);
 
-// Add routes, events & jobs
-if (config.operations.isIndexingModeEnabled) {
-	app.addMethods(path.join(__dirname, 'methods', 'indexer'));
-	app.addEvents(path.join(__dirname, 'events'));
-}
+// Start a temporary broker to query for SDK module names
+// To be used for dynamically registering the available module specific endpoints
+const tempApp = Microservice({
+	...defaultBrokerConfig,
+	name: 'temp_service_indexer',
+	events: {},
+});
+setAppContext(tempApp);
 
-if (config.operations.isDataRetrievalModeEnabled) {
-	app.addMethods(path.join(__dirname, 'methods', 'dataService'));
-	app.addJobs(path.join(__dirname, 'jobs', 'dataService'));
-}
+tempApp.run().then(async () => {
+	const { getRegisteredModules } = require('./shared/constants');
+	const registeredModules = await getRegisteredModules();
+	registeredModules.forEach((module, index, self) => {
+		// Map 'reward' module to the 'dynamicReward' endpoints
+		if (module === 'reward') self[index] = 'dynamicReward';
```

Comment:
May be we can define this in constants file and use it here: 
```
const MODULE = {
	    DYNAMIC_REWARD: 'dynamicReward',
	    REWARD: 'reward',
 };
```
```suggestion
		if (module === MODULE.REWARD) self[index] = MODULE.DYNAMIC_REWARD;
```

Reply:
Applied.